Flaky E2E test: `rendering: plain block buttons` times out on Page.goto (networkidle + external CDNs)

### Summary

The `test-packages-and-docs` CI job intermittently fails on a single Playwright E2E test in `docs/tests/e2e/test_code_blocks.jac` with a `Page.goto` timeout. It is a flaky, network-dependent failure — unrelated to the branch under test.

Example failure: [run 27485664243](https://github.com/jaseci-labs/jaseci/actions/runs/27485664243)

```
FAILED docs/tests/e2e/test_code_blocks.jac::rendering: plain block buttons
E   TimeoutError: Page.goto: Timeout 30000ms exceeded.
  - navigating to "http://127.0.0.1:<port>/_test.html", waiting until "networkidle"
```

Only 1 of 9 tests failed; the same job passes on re-runs and on `main`.

### Root cause

`_with_page()` navigates with the default `wait_until="networkidle"` (`docs/tests/e2e/test_code_blocks.jac:138`). `networkidle` only resolves after 500 ms of zero network activity. But the test HTML fixture (`TEST_HTML`) loads four **external CDN scripts** on every page load:

- `https://unpkg.com/viz.js@2.1.2/viz.js`
- `https://unpkg.com/viz.js@2.1.2/full.render.js`
- `https://cdn.jsdelivr.net/npm/requirejs@2.3.6/require.js`
- `https://cdnjs.cloudflare.com/ajax/libs/monaco-editor/0.52.2/min/vs/loader.min.js`

Each test launches a fresh browser + server, so these are re-fetched cold every time. When any of unpkg/jsdelivr/cdnjs is slow or briefly unreachable on the runner, the network never goes idle and `Page.goto` exceeds the 30 s timeout.

### Suggested fix

The fast rendering tests don't need the CDN scripts to finish loading. Options:

- Switch the rendering tests off `networkidle` to `wait_until="domcontentloaded"` (or `"load"`), or
- Vendor/stub the four CDN scripts locally so the fixture has no external network dependency.

### Workaround

Re-run the failed job — it almost always passes on retry.

--- a/docs/tests/e2e/test_code_blocks.jac
+++ b/docs/tests/e2e/test_code_blocks.jac
@@ -142,8 +142,17 @@ def _stop_server(server: any) {
     }
 }
 
-"""Start server, launch browser, navigate to test page, run callback(page, resp)."""
-def _with_page(callback: any, wait_until: str = "networkidle") {
+"""Start server, launch browser, navigate to test page, run callback(page, resp).
+
+Navigation waits for `domcontentloaded`, not `networkidle`. The fixture's
+`run-code.js` spawns the Pyodide web worker on `DOMContentLoaded`, so the page
+keeps pulling bytes from the CDN long after the DOM is ready and `networkidle`
+(500ms of zero network activity) is never reached within the `goto` timeout —
+flaking the fast rendering tests that don't even need Pyodide. Each test already
+waits on the specific element/condition it depends on (Monaco, Pyodide output,
+Viz SVG, response headers), so none relies on the network going idle.
+"""
+def _with_page(callback: any, wait_until: str = "domcontentloaded") {
     (server, base_url) = _start_server();
     try {
         with sync_playwright() as p {

